### ISSUE TITLE
Add a Lib_PredeployAddresses to replace hardcoded addresses

### ISSUE THREAD

**Issue by smartcontracts** [ethereum-optimism/optimism#895]
**Is your feature request related to a problem? Please describe.**
We have a bunch of hard-coded addresses all over the place. For example:
* https://github.com/ethereum-optimism/optimism/blob/9fea7c94abb5b09a7a0c571370b79e20d7f365c0/packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol#L554
* https://github.com/ethereum-optimism/optimism/blob/9fea7c94abb5b09a7a0c571370b79e20d7f365c0/packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol#L556
* https://github.com/ethereum-optimism/optimism/blob/9fea7c94abb5b09a7a0c571370b79e20d7f365c0/packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol#L832

etc etc

All addresses that start with 0x420.....XX are "predeployed" contracts that are injected into the state when the system is first deployed. [This file](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts/src/predeploys.ts) is essentially the canonical list of these addresses at the moment. I'd like if we moved away from these hard-coded addresses and instead introduced a library that manages these constants.

**Describe the solution you'd like**
`Lib_PredeployAddresses` that looks something like:

```
library Lib_PredeployAddresses {
    address public constant ovmL2ToL1MessagePasser = 0x4200000000000000000000000000000000000000;
    address public constant ovmL1MessageSender = 0x4200000000000000000000000000000000000001;
    .... // etc etc
}
```

I think this library could probably go in a new folder `contracts/optimistic-ethereum/libraries/constants/Lib_PredeployAddresses.sol`

**Describe alternatives you've considered**
Alternatively, we could also just add a `Lib_Constants` that contains *all* of the constants that get used in various places. I'd be down for either approach.

**Comment by rajivpo** [comment]
@smartcontracts I think we can close this out :)

**Comment by smartcontracts** [comment]
Closed by #931. Thanks @rajivpo!
